lib/Dialect/FIRRTL/Transforms/SFCCompat.cpp

```diff
@@ -44,6 +45,16 @@ void SFCCompatPass::runOnOperation() {
 
   bool madeModifications = false;
   SmallVector<InvalidValueOp> invalidOps;
+
+  bool fullAsyncResetExists = false;
+  AnnotationSet::removePortAnnotations(
+      getOperation(), [&](unsigned argNum, Annotation anno) {
+        if (!anno.isClass(fullAsyncResetAnnoClass))
+          return false;
+        fullAsyncResetExists = true;
+        return true;
```

Comment:
Possibly less readable, but I think you can do:

```suggestion
        return fullAsyncResetExists = true;
```